src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -185,6 +201,51 @@ void ProvinceInstance::_update_pops(DefineManager const& define_manager) {
 	}
 }
 
+fixed_point_t ProvinceInstance::calculate_rgo_size(ProductionType const& production_type) const {
+	std::vector<Job> const& jobs = production_type.get_jobs();
+	fixed_point_t total_worker_count_in_province = 0; //not counting equivalents
+	for(Pop const& pop : pops) {
+		bool is_worker_pop_type = false;
+		for(Job const& job : jobs) {
+			if(job.get_pop_type() == pop.get_type()) {
+				is_worker_pop_type = true;
+				break;
+			}
+		}
+
+		if(is_worker_pop_type) {
+			total_worker_count_in_province += pop.get_size();
+		}
+	}
+
+	const fixed_point_t base_workforce_size = production_type.get_base_workforce_size();
+	//TODO: if is_farm add farm size bonus
+	//TODO: if is_mine add mine size bonus
+	return ((total_worker_count_in_province / base_workforce_size).ceil() * fixed_point_t::_1_50()).floor();
+}
+
+void ProvinceInstance::convert_rgo_worker_pops_to_equivalent(ProductionType const& production_type) {
+	std::vector<Job> const& jobs = production_type.get_jobs();
+	fixed_point_t total_worker_count_in_province = 0; //not counting equivalents
+	for(Pop& pop : pops) {
+		bool is_worker_pop_type = false;
```

Comment:
You never change the value of this variable, so the if condition below will never trigger. If this didn't show up in your warnings, you should probably turn it on.

Reply:
What do you mean?
`is_worker_pop_type` is set to true in `if(job.get_pop_type() == pop.get_type())`
`total_worker_count_in_province` is filled in
```cpp
if(is_worker_pop_type) {
	total_worker_count_in_province += pop.get_size();
}
```

Reply:
Right, that's a copy paste mistake from calculate rgo size. THanks for pointing it out. As for warnings, no clue who/what would warn here. Vs code barely helps with cpp programming.

Reply:
Warnings come from your compiler, not your IDE. For example, here is a selection of things you can turn on for clang, if that if your C++ compiler: https://clang.llvm.org/docs/DiagnosticsReference.html